crates/autopilot/src/solvable_orders.rs

```diff
@@ -195,11 +196,11 @@ impl SolvableOrdersCache {
 
         let orders = orders_with_balance(orders, &balances, self.ethflow_contract_address);
         let removed = counter.checkpoint("insufficient_balance", &orders);
-        order_events.extend(removed.into_iter().map(|o| (o, OrderEventLabel::Invalid)));
+        invalid_order_uids.extend(removed);
 
         let orders = filter_dust_orders(orders, &balances, self.ethflow_contract_address);
         let removed = counter.checkpoint("dust_order", &orders);
-        order_events.extend(removed.into_iter().map(|o| (o, OrderEventLabel::Filtered)));
+        filtered_order_events.extend(removed.into_iter().map(|o| (o, OrderEventLabel::Filtered)));
```

Comment:
I like your change to collect `OrderUid`s into different buckets depending on the label.
This can be applied to `filtered_orders` as well to make the code less verbose.